### DISCUSSION TITLE
Is there an example of config Spring Boot with rdf4j server?

### DISCUSSION THREAD

**Discussion by toretest** [eclipse-rdf4j/rdf4j#3371] [Q&A] [answered]
Is there an example of config Spring Boot with rdf4j server?

**Comment by abrokenjester** [comment, marked as answer]
It's not officially released yet, but something along those lines will be part of the upcoming 4.0 release. See https://github.com/eclipse/rdf4j/tree/develop/spring-components/spring-boot-sparql-web .

**Comment by toretest** [comment]
Thanks, I will look into it!

lør. 23. okt. 2021 kl. 01:11 skrev Jeen Broekstra ***@***.***
>:

> It's not officially released yet, but something along those lines will be
> part of the upcoming 4.0 release. See
> https://github.com/eclipse/rdf4j/tree/develop/spring-components/spring-boot-sparql-web
> .
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/eclipse/rdf4j/discussions/3371#discussioncomment-1523739>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AANQRKIZZNPIQBRWC3YSEGLUIHVRXANCNFSM5GRPUHLA>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.
>
>
-- 
Vennlig hilsen

Tore Gard Andersen

**Reply by toretest** [reply]
To test this, is the correct maven config like this:
  <rdf4j.version>4.0.0-SNAPSHOT</rdf4j.version>
  
  
<dependencyManagement>
        <dependencies>
            <dependency>
                <groupId>org.eclipse.rdf4j</groupId>
                <artifactId>rdf4j-bom</artifactId>
                <version>${rdf4j.version}</version>
                <type>pom</type>
                <scope>import</scope>
            </dependency>
        </dependencies>
    </dependencyManagement>

**Reply by abrokenjester** [reply]
Assuming you have configured your local maven project to use our [nightly build snapshot repository ](https://rdf4j.org/download/#source-code-and-nightly-builds), then yes, that should normally work.

_However_, we currently have a build failure in our develop branch, which means that the nightly build for 4.0.0 has not been updated in at least a couple of weeks. It's possible that the latest available snapshot build does not contain the part of the code you need, yet. We're working on a fix for that. 

Alternatively, if you're really adventurous, you can[ build directly from source](https://github.com/eclipse/rdf4j#building-from-source) yourself.

**Comment by fkleedorfer** [comment]
To clarify, @toretest - which of these were you asking about:
1. a spring-boot project that uses an RDF4J Repository as its backend?
2. a spring-boot project that runs an RDF4J Server and offers e.g. a SPARQL interface?

If it's 1., I am working on some user documentation to support people who want to use [rdf4j-spring](https://github.com/eclipse/rdf4j/tree/develop/spring-components/rdf4j-spring), including a demo project, ETA: soon.

**Reply by toretest** [reply]
1.a

Is there a pure spring boot example with a right example pom.

**Reply by fkleedorfer** [reply]
I'll push some code tomorrow

**Reply by fkleedorfer** [reply]
Thanks for the hint, I'll try to put the pom outside the rdf4j pom structure so the example is self contained.

**Reply by fkleedorfer** [reply]
see https://github.com/eclipse/rdf4j/pull/3398

**Reply by fkleedorfer** [reply]
I am interested in any feedback you can give on rdf4j-spring, but especially anything you notice when writing your own DAOs.

**Reply by fkleedorfer** [reply]
@toretest I just finished writing up the user documentation. Apart from the go templates, it's quite readable here:
https://github.com/eclipse/rdf4j/blob/371f297b04963c5fa99c8700e65752a7fc6a35ff/site/content/documentation/programming/spring.md

**Comment by toretest** [comment]
Thanks!

tir. 2. nov. 2021, 23:12 skrev Florian Kleedorfer ***@***.***
